src/conversion/configurationdefinedconverters.js

```diff
@@ -0,0 +1,310 @@
+/**
```

Comment:
This file is named incorrectly. If it doesn't export a variable called `configurationDefinedConverters` then these are 3 separate words and hence the file should be called `configuration-defined-converters.js`.

I'm also unsure about naming this file after "configuration". What if I just want to use these helpers in my feature (which has no configuration)? Can we think about a different name?

Reply:
`definition-based-converters.js`? :D

Reply:
Or we need to find some proper name for these definitions. Something distinctive that we could use all around the code. Conversion specs, conversion maps, conversion rules, conversion... dunno. None of these is better than conversion definitions.

Reply:
Yeah... I go with that `definition-based-converts` and other cleanups. We can review it once again after I make all the requested changes.